code/main.py

```diff
@@ -63,31 +54,52 @@ def main():
     service_principal_id=azure_credentials.get("clientId", "")
     service_principal_password=azure_credentials.get("clientSecret", "")
     subscriptionId=azure_credentials.get("subscriptionId", "")
-    command = ("az login --service-principal --username {APP_ID} --password \"{PASSWORD}\" --tenant {TENANT_ID}").format(
-            APP_ID=service_principal_id, PASSWORD=service_principal_password, TENANT_ID=tenant_id)
-    try:
-        login_result = subprocess.check_output(command, shell=True)
-        print(login_result)
-    except Exception as ex:
-        print(ex)
-        return;
-
+    
     success = False
+    jsonobject = None
     try:
-        jsonobject = None
         with open(template_params_file_path,"r") as f:
             jsonobject = json.load(f);
-        jsonobject["parameters"]["subscriptionID"]["value"] = subscriptionId
-        jsonobject["parameters"]["repo_name"]["value"] = self_repoName
-        jsonobject["parameters"]["pat_token"]["value"] = repo_PatToken
+        jsonobject["subscriptionID"]["value"] = subscriptionId
+        jsonobject["repo_name"]["value"] = self_repoName
+        jsonobject["pat_token"]["value"] = repo_PatToken
         with open(template_params_file_path,"w") as f:
             json.dump(jsonobject,f)
         success = True
     except Exception as ex:
         print("error while updating parameters")
         return;
-    if success:
-        print(deploy_functionApp(template_file_file_path, template_params_file_path, resource_group))
+    credentials=None
+    try:
+        credentials = ServicePrincipalCredentials(
+             client_id=service_principal_id,
+             secret=service_principal_password,
+             tenant=tenant_id
+          )
+    except Exception as ex:
+       print(ex)
+    client = ResourceManagementClient(credentials, subscriptionId)
+    template=None
+    with open(template_file_file_path, 'r') as template_file_fd:
+         template = json.load(template_file_fd)
+    parameters=jsonobject
+    deployment_properties = {
+        'properties':{
+            'mode': DeploymentMode.incremental,
+            'template': template,
+            'parameters': parameters
+        }
+     }
+    try:
+        deployment_async_operation = client.deployments.create_or_update(
```

Comment:
there is option to validate the deployment before creating or updating. 
can you check , using that api as well, before we deploy

Reply:
added validation